### PR TITLE
Feat/mech support

### PR DESCRIPTION
**Pull request by 0xArdi**
This PR adds support in the mech for the changes proposed in #155 .

### CODE REVIEW

File: packages/valory/skills/task_execution/utils/benchmarks.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""Benchmarking for tools."""
+
+import logging
+from typing import Any, Dict, Union
+
+import anthropic
```
Comment (DavidMinarsch) [inline packages/valory/skills/task_execution/utils/benchmarks.py:24]: Why? That seems strange to include here

File: packages/valory/skills/task_execution/utils/benchmarks.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+def encoding_for_model(model: str) -> Encoding:
+    """Get the encoding for a model."""
+    return tiktoken.encoding_for_model(model)
+
+
+def count_tokens(text: str, model: str) -> int:
+    """Count the number of tokens in a text."""
+    if "claude" in model:
+        return anthropic.Anthropic().count_tokens(text)
```
Comment (DavidMinarsch) [inline packages/valory/skills/task_execution/utils/benchmarks.py:40]: The benchmark utils needs to be agnostic to this

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

File: packages/valory/skills/task_execution/utils/benchmarks.py (hunk 1 above)
Comment (0xArdi) [reply packages/valory/skills/task_execution/utils/benchmarks.py:24]: Its copied from here https://github.com/valory-xyz/olas-predict-benchmark/blob/main/benchmark/benchmark/utils.py

**Comment by 0xArdi** [comment]
Req. changes will be addressed in another PR.

**Pull request merged**